src/txadmission.cpp

```diff
@@ -497,8 +524,8 @@ void ThreadTxAdmission()
                     int nDoS = 0;
                     if (state.IsInvalid(nDoS) && state.GetRejectCode() != REJECT_WAITING)
                     {
-                        LOG(MEMPOOL, "%s from peer=%s was not accepted: %s\n", tx->GetHash().ToString(), txd.nodeName,
-                            FormatStateMessage(state));
+                        LOG(MEMPOOL, "%s from peer=%s was not accepted: %s\ntx: %s", tx->GetHash().ToString(),
+                            txd.nodeName, FormatStateMessage(state), EncodeHexTx(*tx));
```

Comment:
Can a remote peer send you loads of big, but rejected transactions without getting DoS points? If so, `EncodeHexTx` may be a CPU hog.

Reply:
LOG is a macro that expands its printf args inside the if that determines whether the log is on or not.  So it only serializes the tx if MEMPOOL logging is on.